### PR TITLE
Ensure HEIC files selectable from “Upload” button

### LINKED ISSUE #66291: Chrome: Unable to upload HEIC image through post editor
### Description

Using Chrome's built-in upload functionality when inserting an HEIC image with the post editor, files of type HEIC are disabled for selection. However, files can be added through drag-n-drop on the Image block, or added through the media library either with the upload or multi-file feature. This issue occurs regardless of server-side HEIC support (i.e. availability of ImageMagick.)

Originally reported at https://github.com/WordPress/twentytwentyfive/issues/583, then https://core.trac.wordpress.org/ticket/62268 for tracking with the 6.7 release. This issue is being reported upstream to GB.

### Expected behavior
Though Chrome cannot display HEIC natively, it should be an allowed filetype in the browser uploader. This is consistent with [6.7 support for this media type](https://make.wordpress.org/core/2024/08/15/automatic-conversion-of-heic-images-to-jpeg-in-wordpress-6-7/).

### Step-by-step reproduction instructions

1. In Chrome, add a new post.
2. Add an Image block, and select "Upload".
3. 🐞 Observe that HEIC images cannot be selected for upload.

### Screenshots, screen recording, code snippet

HEIC images are not selectable in upload dialog in Chrome

### Environment info

- Hardware: MacBook Pro Apple M1 Pro
- OS: macOS 14.6.1
- Browser: Google Chrome 130.0.6723.59 (also confirmed working as expected on Safari 17.6, Mozilla Firefox 131.0.3)
- Tested in local install and Playground
- WordPress 6.7 `trunk` / GB not installed

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### PR DESCRIPTION
**Pull request by adamsilverstein**
Fixes https://github.com/WordPress/gutenberg/issues/66291.

This may be an upstream Chrome bug, see https://core.trac.wordpress.org/ticket/62268

### CODE REVIEW

**Comment by ironprogrammer** [comment]
Thank you for the PR, @adamsilverstein!

In my testing, this resolves the issue, and allows HEIC files to be selected for upload on the post screen. ✅

**Comment by swissspidy** [comment]
For testing, what if the server doesn‘t support HEIC?

**Comment by ndiego** [comment]
> For testing, what if the server doesn‘t support HEIC?

Yeah, while this fix addresses the situation where an HEIC image is not selectable, there is still an issue with HEIC images being uploadable when they shouldn't be. 

In the video below, I tested in Playground, which does not include `ImageMagick`. In this situation, I don't think you should be able to upload an HEIC image. **Now, it's important to note that this behavior currently exists in 6.6.** 

So, it's not a regression, but it's still an unfortunate bug that will become more noticeable once 6.7 is out and people start trying to use HEIC images when they are not supported. 

https://github.com/user-attachments/assets/251375ee-2ffb-4ce8-9142-2e01fe5ca6b9

**Comment by afercia** [comment]
Related: https://github.com/WordPress/gutenberg/issues/66293

**Comment by Mamaduka** [comment]
Instead of updating all individual uses, maybe we should enhance the `FormFileUpload` component when the `accept` prop contains `image/*`. I believe all `accept` props from different components are passed down to it at the end.

File: packages/components/src/form-file-upload/index.tsx [hunk 1]
```diff
@@ -18,7 +18,7 @@ import type { FormFileUploadProps } from './types';
  *
  * const MyFormFileUpload = () => (
  *   <FormFileUpload
- *     accept="image/*"
+ *     accept="image/*, , image/heic"
```
Comment (Mamaduka) [inline packages/components/src/form-file-upload/index.tsx]:
```suggestion
 *     accept="image/*, image/heic"
```

Typo.

**Comment by adamsilverstein** [comment]
> Instead of updating all individual uses, maybe we should enhance the `FormFileUpload` component when the `accept` prop contains `image/*`. I believe all `accept` props from different components are passed down to it at the end.

That makes sense and it would be easier to revert if/when Chrome lands the fix in stable as well. I looked at that originally, but since FormFileUpload could be used elsewhere there would need to be added logic to only add the type when uploading images as you said, adding some complexity.

I can rework with that approach.

**Comment by adamsilverstein** [comment]
> In the video below, I tested in Playground, which does not include ImageMagick. In this situation, I don't think you should be able to upload an HEIC image. Now, it's important to note that this behavior currently exists in 6.6.

You shouldn't be able to upload HEIC files when the server doesn't support them. This is a completely separate issue from being able to select HEIC images in the file upload button though. 

It looks like core media handles this correctly, but not in Gutenberg. This issue should address that: https://github.com/WordPress/gutenberg/issues/66293

File: packages/components/src/form-file-upload/index.tsx [hunk 2]
```diff
@@ -18,7 +18,7 @@ import type { FormFileUploadProps } from './types';
  *
  * const MyFormFileUpload = () => (
  *   <FormFileUpload
- *     accept="image/*"
+ *     accept="image/*, image/heic"
```
Comment (swissspidy) [inline packages/components/src/form-file-upload/index.tsx]: I don't think we should change this example code here

**Comment by Mamaduka** [comment]
@adamsilverstein, a bit of naive logic below, but I think it should do the trick.

```js
// @todo: Explain the need for the `image/heic` type and link the Chrome bug.
const compatAccept = !! accept?.includes( 'image/*' )
	? `${ accept }, image/heic`
	: accept;
```

File: packages/components/src/form-file-upload/index.tsx (hunk 2 above)
Comment (Mamaduka) [reply packages/components/src/form-file-upload/index.tsx]: Right, this won't be necessary after the `FormFileUpload` component update.

**Comment by swissspidy** [comment]
This should probably also add `image/heif` btw

**Comment by adamsilverstein** [comment]
> This should probably also add image/heif btw

We can, although I have yet to locate an image with the mime type `image/heif`...

**Comment by swissspidy** [comment]
There was just one added to https://core.trac.wordpress.org/ticket/62272 yesterday

**Comment by adamsilverstein** [comment]
Note: this has been fixed upstream in Chromium; I'm not sure how long that fix takes to make it to the Stable channel.

https://chromium-review.googlesource.com/c/chromium/src/+/5967494

**Review by Mamaduka — APPROVED** [review]

**Comment by getdave** [comment]
Testing this now with a view to bring into the packages release for the final RC3 of WP 6.7.

**Review by getdave — APPROVED** [review]
I tested this and confirmed that this allows selection of HEIC images. The rationale for the fix is good and I think we should bring this into WordPress 6.7.

**Pull request merged**